modules/lib/mabAddressbookUtil.js

```diff
@@ -56,5 +59,117 @@ var AddressbookUtil = {
 
       foStream.close();
     }
+  },
+
+  /**
+   * Ask the user to point to a vCard and import the contacts into the database
+   *
+   * @param {Addressbook} Addressbook to add the imported contacts to
+   * @returns {Promise[]} Array of promises for adding the new contacts
+   */
+  importContacts: function(addressbook) {
+
+    var filePickerInterface = Components.interfaces.nsIFilePicker;
+    var filePicker = Components.classes["@mozilla.org/filepicker;1"].createInstance(filePickerInterface);
+    var windowMediator = Cc["@mozilla.org/appshell/window-mediator;1"].getService(Ci.nsIWindowMediator);
+    var window = windowMediator.getMostRecentWindow(null);
+
+    filePicker.init(window, "Load contacts from", filePickerInterface.modeOpen);
+    filePicker.appendFilter("vCard", "*.vcf; *.vcard");
+    filePicker.appendFilters(filePickerInterface.filterAll);
+    filePicker.filterIndex = 0;
+    filePicker.defaultString = "contacts.vcf";
+
+    var returnValue = filePicker.show();
+
+    if (returnValue == filePickerInterface.returnOK) {
+
+      var inputStream = Components.classes["@mozilla.org/network/file-input-stream;1"]
+        .createInstance(Components.interfaces.nsIFileInputStream);
+
+      inputStream.init(filePicker.file, 0x01, 0444, 0); // readonly
+
+      var fileContents = NetUtil.readInputStreamToString(
+          inputStream,
+          inputStream.available(),
+          {});
+
+      inputStream.close();
+
+      var contacts = ICAL.parse(fileContents);
+      // detect error from parsing
+      if (!contacts) {
+        throw "Invalid vCard";
+      }
+
+      // check if it is an array of vcards or a single vcard
+      if (contacts[0] == 'vcard') {
+        // make a single contact a list to work with the next section
+        contacts = [contacts];
+      }
+
+      // convert jcards into Contact objects
+      var contactPromises = contacts.map(function(vcard) {
+        var contact =  new ICAL.Component(vcard);
+
+        // try to get the name of the contact
+        var name = contact.getFirstPropertyValue("fn");
```

Comment:
Depending on the vCard version, it either has to have fn formatted name (3.0/4.0) or n name (2.1, 3.0)
So it should do a check to see if fn is present and then n. 
if neither, its not a valid vcard format but we should still import it right?
so, then use an email and if no email then ??? - ask guenter